buildSrc/src/main/java/gradlebuild/JniNokeePlugin.java

```diff
@@ -163,6 +209,10 @@ private void configureNativeVersionGeneration(Project project) {
             task.getGeneratedJavaSourcesDir().set(new File(generatedFilesDir, "version/java"));
             task.getVersionClassPackageName().set(nativeVersion.getVersionClassPackageName());
             task.getVersionClassName().set(nativeVersion.getVersionClassName());
+            task.getNativeSources().from(
+                library(project).map(JavaNativeInterfaceLibraryUtils::cppSources),
+                library(project).map(JavaNativeInterfaceLibraryUtils::privateHeaders)
+            );
```

Comment:
❓ The native sources no longer include the auto-generated JNI headers as they were causing a cycle in the task graph. In theory, the JNI headers are implicitly covered by the C++ sources given they implement the prototype hence have the same signature. It's up to you to validate if this change is correct or not.